packages/eui/src/components/datagrid/body/data_grid_body_custom.tsx

```diff
@@ -26,178 +21,236 @@ import { useDataGridHeader } from './header';
+          renderFooterCellValue,
+          renderCellPopover,
+          rowIndex: visibleRows.visibleRowCount,
+          visibleRowIndex: visibleRows.visibleRowCount,
+          interactiveCellId,
+          leadingControlColumns,
+          trailingControlColumns,
+          columns,
+          columnWidths,
+          defaultColumnWidth,
+          schema,
+        }),
+        [
+          renderFooterCellValue,
+          renderCellPopover,
+          visibleRows.visibleRowCount,
+          interactiveCellId,
+          leadingControlColumns,
+          trailingControlColumns,
+          columns,
+          columnWidths,
+          defaultColumnWidth,
+          schema,
+        ]
+      );
 
-  // Allow consumers to pass custom props/attributes/listeners etc. to the wrapping div
-  const [customGridBodyProps, setCustomGridBodyProps] =
-    useState<EuiDataGridSetCustomGridBodyProps>({});
-
-  return (
-    <div
-      {...customGridBodyProps}
-      className={classNames(
-        'euiDataGrid__customRenderBody',
-        customGridBodyProps?.className
-      )}
-    >
-      {headerRow}
-      {renderCustomGridBody!({
-        visibleColumns,
-        visibleRowData: visibleRows,
-        Cell,
-        setCustomGridBodyProps,
-      })}
-      {footerRow}
-    </div>
+      const { footerRow } = useDataGridFooter(footerRowProps);
+
+      /**
+       * Cell render fn
+       */
+      const cellProps = useMemo(() => {
+        return {
+          schema,
+          schemaDetectors,
+          pagination,
+          columns,
+          leadingControlColumns,
+          trailingControlColumns,
+          visibleColCount,
+          columnWidths,
+          defaultColumnWidth,
+          renderCellValue,
+          cellContext,
+          renderCellPopover,
+          interactiveCellId,
+          setRowHeight,
+          rowHeightsOptions,
+          rowHeightUtils,
+        };
+      }, [
+        schema,
+        schemaDetectors,
+        pagination,
+        columns,
+        leadingControlColumns,
+        trailingControlColumns,
+        visibleColCount,
+        columnWidths,
+        defaultColumnWidth,
+        renderCellValue,
+        cellContext,
+        renderCellPopover,
+        interactiveCellId,
+        setRowHeight,
+        rowHeightsOptions,
+        rowHeightUtils,
+      ]);
+
+      const Cell = useMemo<EuiDataGridCustomBodyProps['Cell']>(
+        () =>
+          ({ colIndex, visibleRowIndex, ...rest }) => {
+            const style = {
+              height: rowHeightUtils.isAutoHeight(
+                visibleRowIndex,
+                rest.rowHeightsOptions ?? rowHeightsOptions
+              )
+                ? 'auto'
+                : getRowHeight(visibleRowIndex),
+            };
+
+            const props = {
+              colIndex,
+              visibleRowIndex,
+              style,
+              ...cellProps,
+            };
+            return <CellWrapper {...props} {...rest} />;
```

Comment:
@cee, do you think it should be changed as below : 
```suggestion
            return <CellWrapper {...rest} {...props} />;
```

Reply:
I'm fine leaving the `...rest` afterwards - since it's a custom renderer in any case, we should be allowing anything that the consumer sets as a cell prop override whatever they need. 🤷

Reply:
Makes sense. Thanks.